subsystem/apiclient/kubeapi.go

```diff
@@ -0,0 +1,288 @@
+package apiclient
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"github.com/go-openapi/strfmt"
+	"github.com/go-openapi/swag"
+	"github.com/google/uuid"
+	"github.com/jinzhu/gorm"
+	"github.com/openshift/assisted-service/client/installer"
+	"github.com/openshift/assisted-service/internal/common"
+	adiiov1alpha1 "github.com/openshift/assisted-service/internal/controller/api/v1alpha1"
+	"github.com/openshift/assisted-service/models"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+)
+
+const (
+	APIVersion = "adi.io.my.domain/v1alpha1"
+)
+
+type KubeAPIClient struct {
+	client    runtimeclient.Client
+	namespace string
+	db        *gorm.DB
+}
+
+func NewKubeAPIClient(
+	namespace string,
+	config *rest.Config,
+	db *gorm.DB,
+) (*KubeAPIClient, error) {
+
+	client, err := runtimeclient.New(config, runtimeclient.Options{Scheme: scheme.Scheme})
+	if err != nil {
+		return nil, err
+	}
+	kc := &KubeAPIClient{
+		client:    client,
+		namespace: namespace,
+		db:        db,
+	}
+	return kc, nil
+}
```

Comment:
What is the subsystem layer in your code base?

Curious because your controllers have a client already, so I wasn't sure what the purpose of creating another was with this utility.

Reply:
It's a test suite whose scope is bigger than unit tests but smaller than e2e tests, generally targeting one "subsystem" at a time.